lib/src/multipart_request.dart

```diff
@@ -44,10 +44,14 @@ class MultipartRequest extends BaseRequest {
   /// The private version of [files].
   final List<MultipartFile> _files;
 
+  /// The boundary prefix. When not set, defaults to "dart-http-boundary-".
+  String boundaryPrefix;
```

Comment:
Make this final

Reply:
Consider the following:
~~~dart
      var request = MultipartRequest(method, Uri.parse(url));
      request.boundaryPrefix = "custom-boundary-";
~~~
When making it final, the compiler says: 'boundaryPrefix' can't be used as a setter because it is final.


How about a setter and private backing field?
~~~dart
  /// The boundary prefix. When not set, defaults to "dart-http-boundary-".
  set boundaryPrefix(String value) => _boundaryPrefix = value;
  String _boundaryPrefix = "dart-http-boundary-";

  /// Creates a new [MultipartRequest].
  MultipartRequest(String method, Uri url)
      : fields = {},
        _files = <MultipartFile>[],
        super(method, url);
~~~

Reply:
How about aiming for a usage like:

```dart
var request = MultipartRequest(method, Uri.parse(url), boundaryPrefix: "custom-boundary-");
```

```dart
final String boundaryPrefix;

MultipartRequest(String method, Uri url, {String boundaryPrefix})
    : fields = {},
      _files = <MultipartFile>[],
      boundaryPrefix = boundaryPrefix ?? "dart-http-boundary-",
      super(method, url);
```

Reply:
> How about aiming for a usage like:
> 
> ```dart
> var request = MultipartRequest(method, Uri.parse(url), boundaryPrefix: "custom-boundary-");
> ```
> 
> ```dart
> final String boundaryPrefix;
> 
> MultipartRequest(String method, Uri url, {String boundaryPrefix})
>     : fields = {},
>       _files = <MultipartFile>[],
>       boundaryPrefix = boundaryPrefix ?? "dart-http-boundary-",
>       super(method, url);
> ```

I can imagine not many users would use this adjustable property, that's why I was avoiding the population of the ctor.

Let me implement your example.